### DISCUSSION TITLE
Can't use getUserWorldPosition() in fragment block

### DISCUSSION THREAD

**Discussion by noergel1** [google/filament#6723] [Q&A]
I'm implementing triplanar mapping, but the texture is basically "following" the camera. So either the world-position or world-normal is changing with the view matrix.
I am not sure if thats the reason for it, but I can't use the `getUserWorldPosition()` function, and had to resort to `getWorldOffset()` or `getWorldPosition()`. The Filament version is 1.31.2.

Compiling the shader with `getWorldPosition()` results in

> ERROR: triplanarMapping.mat:34: 'getUserWorldPosition' : no matching overloaded function found

My shader code looks like this:

```
material {

	name : "Triplanar Texture Mapping",
	parameters : [
		{
			type : sampler2d,
			name : albedo
		}
	],
	requires : [
	
	],
	shadingModel : lit,
	blending : opaque
}

fragment {
	float4 SampleTextureThrice(float3 _wPos, float3 _wNorm) {
		float scale = 0.25f;
	
		vec3 blending = abs( _wNorm );
		blending = normalize(max(blending, 0.00001)); // Force weights to sum to 1.0 
		float b = (blending.x + blending.y + blending.z);
		blending /= vec3(b, b, b);
		
		vec4 xaxis = texture( materialParams_albedo, _wPos.yz * scale);
		vec4 yaxis = texture( materialParams_albedo, _wPos.xz * scale);
		vec4 zaxis = texture( materialParams_albedo, _wPos.xy * scale);
		// blend the results of the 3 planar projections. 
		return (xaxis * blending.x + xaxis * blending.y + zaxis * blending.z);
	}
	
	void material(inout MaterialInputs material) {
		prepareMaterial(material);
		material.baseColor = SampleTextureThrice(getWorldOffset(), getWorldNormalVector());
	}
}
```

Thanks in advance for your help!

Edit: I just read in the material guide that the world pos is shifted by the view pos, so thats the reason for the textures following. Does anyone know why I cant use the `getUserWorldPosition` function?

**Comment by bejado** [comment]
`getUserWorldPosition()` was added in `v1.32.0`, so you'll need to upgrade.
